compiler/rustc_resolve/src/macros.rs

```diff
@@ -217,25 +217,22 @@ impl<'a> ResolverExpand for Resolver<'a> {
         features: &[Symbol],
         parent_module_id: Option<NodeId>,
     ) -> ExpnId {
+        let (parent_scope, parent_module) = if let Some(module_id) = parent_module_id {
+            let parent_def_id = self.local_def_id(module_id);
+            (self.module_map[&parent_def_id], parent_def_id)
+        } else {
+            (self.empty_module, CRATE_DEF_ID)
+        };
+
         let expn_id = ExpnId::fresh(Some(ExpnData::allow_unstable(
             ExpnKind::AstPass(pass),
             call_site,
             self.session.edition(),
             features.into(),
             None,
+            Some(parent_module.to_def_id()),
```

Comment:
This logic is suspicious (that's pre-existing).
I need to check what happens if we are 1) always passing `None` here or 2) passing `None` in the `parent_module_id.is_none()` case.
Right now the code seems to conflate `parent_module: None` with `parent_module: Some(CRATE_DEF_ID)`.
However, it's possible that they should always be conflated and `parent_module: Option<DefId>` should be turned into `parent_module: DefId`.

Reply:
Changing this logic makes some proc macro tests fail. I have not investigated yet.

Reply:
Could you follow the commit linked from https://github.com/rust-lang/rust/pull/86694#discussion_r663559543 here and pass `None` if the AST pass doesn't have a parent module?